yin_yang/plugins/_plugin.py

```diff
@@ -253,25 +253,45 @@ def set_theme(self, theme: str):
 
 
 class DBusPlugin(Plugin):
-    def __init__(self):
+    """A class for plugins that mainly switching theme via DBus"""
+    def __init__(self, message_data: List[str]):
         super().__init__()
         self.connection = QDBusConnection.sessionBus()
+        self.message: QDBusMessage
+        self.message_data: List[str] = message_data  # Store DBusMessage data(destination, path, interface, method)
 
     def set_theme(self, theme: str):
+        """Check arguments, create DBus message and then call"""
         if not (self.available and self.enabled):
             return
 
         if not theme:
             raise ValueError(f'Theme \"{theme}\" is invalid')
 
-        self.call(self.create_message(theme))
+        self.create_message(theme)
+        self.call()
 
     @abstractmethod
-    def create_message(self, theme: str) -> QDBusMessage:
+    def create_message(self, theme: str) -> None:
```

Comment:
Is there a particular reason why we change the signature?

Reply:
Because I save the dbus message in a member variable so I think it is no need to return something. Is this a bad change?

Reply:
I understand that you save the message in the member variable, but I don't understand why. I think this is a bit harder to read, as it's harder to follow the "data flow". Might be just my taste, though. If this has no benefit for functionality, please revert that.

Reply:
I think you are right, editing...

Reply:
Done.